pkg/apiserver/apis/v1alpha3/ipblockusage_types.go

```diff
@@ -0,0 +1,68 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GroupName    = "nsx.vmware.com"
+	Version      = "v1alpha3"
+	ResourceName = "ipblockusages"
+)
+
+const (
+	External Visibility = "EXTERNAL"
+	Project  Visibility = "PROJECT"
+	Private  Visibility = "PRIVATE"
+)
+
+type Visibility string
+
+type IPAddress struct {
+	Address string `json:"address,omitempty" protobuf:"bytes,1,opt,name=address"`
+	Subnet  string `json:"subnet,omitempty" protobuf:"bytes,2,opt,name=subnet"`
+}
+
+type AllocatedByVPC struct {
+	Count       int         `json:"count,omitempty" protobuf:"bytes,1,opt,name=count"`
+	IPAddresses []IPAddress `json:"ipAddresses,omitempty" protobuf:"bytes,2,opt,name=ipAddresses"`
+	CIDR        string      `json:"cidr,omitempty" protobuf:"bytes,3,opt,name=cidr"`
+	Visibility  Visibility  `json:"visibility,omitempty" protobuf:"bytes,4,opt,name=visibility"`
+}
+
+// IPBlocks contains the used IPAddresses and CIDR.
+type IPBlocks struct {
+	Path      string `json:"path,omitempty" protobuf:"bytes,1,rep,name=path"`
+	Available int    `json:"available,omitempty" protobuf:"bytes,2,rep,name=available"`
```

Comment:
There is no "available count" per Block? It does not make sense to have a single available count given each Block has different visibilities. Please monitor NSX API changes.

Reply:
Yes, that is true. From NSX IPBlock API, there should only be Path and CIDR fields. 

Here, the IPBlocks definition just an internal data type specific for IPBlockUsage to align
NSX VPC IPBlockUsage API:

 GET /policy/api/v1/orgs/default/projects/project-1/vpcs/vpc-1/ip-address-usage
  example_response: |
    {
    "ip_blocks": [
        {
            "path": "/infra/ip-blocks/934e516a-733d-47c6-bd82-738b4fa69a9c",
            "available": 1024,
            "allocated_by_vpc": {
                "count": 512,
                "ip-addresses": [
                    {
                        "address": "10.0.0.1-10.0.0.4",
                        "subnet": "/orgs/default/projects/<project-id>/vpcs/vpc_1/subnets/subnet_1"
                    }
                ],
                "cidr": "10.0.0.0/16",
                "visibility": "EXTERNAL"
            }
        }
    ]
    }
    
In this API, is to retrives the usage information for IP addresses within a specific VPC. The response will tell user each IPBlock info, including available cout, allocated ipaddress by which subnet, cidr, and visibilities inside one VPC.

Reply:
Ok, I think the confusion is your struct is named "IPBlocks", but it is actually for a single IPBlock info. Please use a better name. What is the type name in NSX API?

Reply:
I see. It should be IPBlock for a single IPBlock.
in NSX API: ip_blocks is a list for each single ip_block. I will rename my code to align this.